scripts/zones/AlTaieu/npcs/qm3.lua

```diff
@@ -4,34 +4,29 @@
 -- Allows players to spawn the Jailer of Prudence by trading the Third Virtue, Deed of Sensibility, and High-Quality Hpemde Organ to a ???.
 -- !pos , 706 -1 22
 -----------------------------------
-local ID = require("scripts/zones/AlTaieu/IDs");
+local ID = require("scripts/zones/AlTaieu/IDs")
+require("scripts/globals/npc_util")
 -----------------------------------
 
 function onTrade(player,npc,trade)
-    -- JAILER OF PRUDENCE
-    if (
-        not GetMobByID(ID.mob.JAILER_OF_PRUDENCE_1):isSpawned() and
-        not GetMobByID(ID.mob.JAILER_OF_PRUDENCE_2):isSpawned() and
-        trade:hasItemQty(1856,1) and -- third_virtue
-        trade:hasItemQty(1870,1) and -- deed_of_sensibility
-        trade:hasItemQty(1871,1) and -- high-quality_hpemde_organ
-        trade:getItemCount() == 3
-    ) then
-        player:tradeComplete();
-        SpawnMob(ID.mob.JAILER_OF_PRUDENCE_1):updateClaim(player); -- Spawn Jailer of Prudence 1
-        SpawnMob(ID.mob.JAILER_OF_PRUDENCE_2);                     -- Spawn Jailer of Prudence 2 unclaimed
+        if npcUtil.tradeHas(trade, {1856, 1870, 1871}) and npcUtil.popFromQM(player, npc, ID.mob.JAILER_OF_PRUDENCE_1) then
+        player:confirmTrade()
+        SpawnMob(ID.mob.JAILER_OF_PRUDENCE_1):updateClaim(player)
```

Comment:
And here.  In this case, if you want to set claim for some spawned mobs, and not for others, then set the claim argument on popFromQM to false, then updateClaim (but don't SpawnMob) inside the conditional block for the ones you want to have aggro.